MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -183,108 +170,97 @@ public void UpdateEncoderPath(string path, string pathType)
-                        var config = ConfigurationManager.GetConfiguration<EncodingOptions>("encoding");
-                        config.EncoderAppPathCustom = FFmpegPath;
-                        ConfigurationManager.SaveConfiguration("encoding", config);
+                    // Write the validated mpeg path to the xml as <EncoderAppPathCustom>
+                    // This ensures its not lost on new startup
+                    var config = ConfigurationManager.GetConfiguration<EncodingOptions>("encoding");
+                    config.EncoderAppPathCustom = FFmpegPath;
+                    ConfigurationManager.SaveConfiguration("encoding", config);
 
-                        FFprobePath = null; // Clear probe path so it gets relooked in ReInit()
+                    FFprobePath = null; // Clear probe path so it gets relooked in ReInit()
 
-                        ReInit();
-                    }
+                    ReInit();
                 }
             }
         }
 
-        private bool ValidatePath(string type, string path)
+        /// <summary>
+        /// Validates the supplied FQPN to ensure it is a FFxxx utility.
+        /// If checks pass, global variable FFmpegPath (or FFprobePath) and
+        /// EncoderLocation (or ProbeLocation) are updated.
+        /// </summary>
+        /// <param name="type">Either mpeg or probe</param>
+        /// <param name="path">FQPN to test</param>
+        /// <param name="location">Location (External, Custom, System) of tool</param>
+        /// <returns></returns>
+        private bool ValidatePath(FFtype type, string path, FFmpegLocation location)
         {
+            bool rc = false;
+
             if (!string.IsNullOrEmpty(path))
             {
                 if (File.Exists(path))
                 {
-                    var valid = new EncoderValidator(_logger, _processFactory).ValidateVersion(path, true);
+                    rc = new EncoderValidator(_logger, _processFactory).ValidateVersion(path, false);
 
-                    if (valid == true)
+                    // Only update the global variables if the checks passed
+                    if (rc)
                     {
-                        return true;
+                        if (type == FFtype.mpeg)
+                        {
+                            FFmpegPath = path;
+                            EncoderLocation = location;
+                        }
+                        else
+                        {
+                            FFprobePath = path;
+                            ProbeLocation = location;
+                        }
                     }
                     else
-                    {
-                        _logger.LogError("{0}: Failed validation checks.  Version 4.0 or greater is required: {1}", type, path);
+                    { 
+                        _logger.LogError("FFmpeg: {0}: Failed version check: {1}", location.ToString(), path);
```

Comment:
This log is now wrong about stating `FFmpeg` (as you now use same function for `ffmpeg` and `ffprobe`). Log messages have to be updated.